python/tank/templatekey.py

```diff
@@ -292,31 +304,164 @@ def __init__(self,
                                          abstract=abstract,
                                          length=length)
 
-        if not(format_spec is None or isinstance(format_spec, basestring)):
-            msg = "Format_spec for TemplateKey %s is not of type string: %s"
+    def _init_format_spec(self, name, format_spec):
+        """
+        Asserts that the format_spec parameter is a valid value.
+
+        :param name: Name of this template key.
+        :param format_spec: Parameter to be validated.
+
+        :raises TankError: Raised when the parameter is not a string that maching a %d format
+                           option.
+        """
+        # No format spec means no formatting options.
+        if format_spec is None:
+            self._padding_char = None
+            self._minimum_width = None
+            self.format_spec = None
+            return
+
+        if isinstance(format_spec, basestring) is False:
```

Comment:
It just came out that way. Kinda out of character for me and our code. I'll fix it.